src/components/ExtraIcon.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <a
+    :class="{ 'extra-icon': true, 'extra-icon-off': !this.on }"
+    @click="handleClick($event)"
+  >
+    <slot />
+  </a>
+</template>
+
+<script>
+export default {
+  props: {
+    on: { type: Boolean, default: true }
+  },
+  methods: {
+    handleClick(e) {
```

Comment:
You could move all the logic from here: https://github.com/balancer-labs/snapshot/pull/69/files#diff-a3c1ef62355391e980435ca4d7ea595bR62-R71 to this component so we can use the Favorite icon in others places and reduce logic in the homepage. And maybe change all the name from "ExtraIcon" to "Button/Favorite".

Reply:
With the last commit in which I added icons I went with a bit different route where this component became a `StatefulIcon` which can be toggled between two states (so it can handle appearing on hover with a particular state).
With that `StatefulIcon` I don't think there is a much sense to extract this component to `FavoriteIcon` as doesn't add any huge difference over just using `StatefulIcon`. I pushed a commit after your comment so the diff you posted got shifted and I'm not sure what you are referring to.

Reply:
I was referring to this https://github.com/balancer-labs/snapshot/pull/69/files#diff-a3c1ef62355391e980435ca4d7ea595bR65-R75

Reply:
I tend to prefer smart component and dumb view instead of the opposite. I'm fine either ways in this case it wont change much.